src/timearray.jl

```diff
@@ -2,30 +2,36 @@
 
 import Base: convert, length, show, getindex, start, next, done, isempty, endof
 
-abstract AbstractTimeSeries
+abstract type AbstractTimeSeries end
 
-immutable TimeArray{T, N, D<:TimeType, A<:AbstractArray} <: AbstractTimeSeries
+struct TimeArray{T, N, D <: TimeType, A <: AbstractArray{T, N}} <: AbstractTimeSeries
 
     timestamp::Vector{D}
     values::A
     colnames::Vector{String}
     meta::Any
 
-    function TimeArray(timestamp::AbstractVector{D},
-                       values::AbstractArray{T,N},
-                       colnames::Vector{String},
-                       meta::Any)
-                           nrow, ncol = size(values, 1), size(values, 2)
-                           nrow != size(timestamp, 1) ? error("values must match length of timestamp"):
-                           ncol != size(colnames,1) ? error("column names must match width of array"):
-                           # length(unique(colnames)) != ncol ? colnames = replace_dupes(colnames):
-                           timestamp != unique(timestamp) ? error("there are duplicate dates"):
-                           ~(flipdim(timestamp, 1) == sort(timestamp) || timestamp == sort(timestamp)) ? error("dates are mangled"):
-                           flipdim(timestamp, 1) == sort(timestamp) ?
-                           new(flipdim(timestamp, 1), flipdim(values, 1), replace_dupes(colnames), meta):
-                           #new(flipdim(timestamp, 1), flipdim(values, 1), colnames, meta):
-                           new(timestamp, values, replace_dupes(colnames), meta)
-                           #new(timestamp, values, colnames, meta)
+    function TimeArray{T, N, D, A}(
+            timestamp::AbstractVector{D},
+            values::A,
+            colnames::Vector{String},
+            meta::Any) where {T, N, D <: TimeType, A <: AbstractArray{T, N}}
+        nrow, ncol = size(values, 1, 2)
+
+        if nrow != size(timestamp, 1)
```

Comment:
the history behind this was to avoid `length` in favor of `size` for consistent syntax to follow DataFrames. I think I originally had `length`. In any case, I agree with using `length`.